src/core/tcp.c

```diff
@@ -1225,8 +1265,8 @@ tcp_slowtmr(void)
     LWIP_ASSERT("tcp_slowtmr: active pcb->state != CLOSED\n", pcb->state != CLOSED);
     LWIP_ASSERT("tcp_slowtmr: active pcb->state != LISTEN\n", pcb->state != LISTEN);
     LWIP_ASSERT("tcp_slowtmr: active pcb->state != TIME-WAIT\n", pcb->state != TIME_WAIT);
-    if (pcb->last_timer == tcp_timer_ctr) {
-      /* skip this pcb, we have already processed it */
+    if ((pcb->last_timer == tcp_timer_ctr) || !tcp_trylock(pcb)) {
+      /* skip this pcb, either we have already processed it, or it cannot be locked without risking deadlock */
```

Comment:
Is the path of recourse for a failed trylock to just wait for the next timer expiry? If so, could this cause any side-effects?

Reply:
Yes, the path of recourse for a failed trylock is to wait for the next timer expiry. As far I can see, there shouldn't be side-effects other than possibly delaying packet re-transmissions (which would be analogous to other network delays) or keeping a connection in a given state (e.g. FIN-WAIT-2) for longer than normal. Also, most of the processing done by tcp_slowtmr() is relevant for TCP connections that have been idle for some time, and since a failed trylock likely means data is currently being sent/received on a TCP connection, it seems unlikely that such connection would need processing by tcp_slowtmr().